filters/auth/grantflowstate.go

```diff
@@ -0,0 +1,132 @@
+package auth
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/zalando/skipper/secrets"
+)
+
+const randomStringLength = 20
+
+type state struct {
+	Rand       string `json:"rand"`
+	Validity   int64  `json:"validity"`
+	Nonce      string `json:"nonce"`
+	RequestURL string `json:"redirectUrl"`
+}
+
+type flowState struct {
+	secrets     *secrets.Registry
+	secretsFile string
+}
+
+var errExpiredAuthState = errors.New("expired auth state")
+
+const (
+	secretSize    = 20
+	letterBytes   = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+	letterIdxBits = 6                    // 6 bits to represent a letter index
+	letterIdxMask = 1<<letterIdxBits - 1 // All 1-bits, as many as letterIdxBits
+	letterIdxMax  = 63 / letterIdxBits   // # of letter indices fitting in 63 bits
+)
+
+var (
+	src = rand.NewSource(time.Now().UnixNano())
+)
+
+// https://stackoverflow.com/questions/22892120/how-to-generate-a-random-string-of-a-fixed-length-in-golang
```

Comment:
This is a famous answer but I would go with
> Paul's solution provides a simple, general solution.

IMO the performance difference does not make sense in the context of back-and-forth redirects while readability suffers

Reply:
This code was removed as a result of removing the `randString()` function.